packages/@n8n/task-runner-node-js/src/start.ts

```diff
@@ -1,3 +1,34 @@
+import * as a from 'node:assert/strict';
+
 import { JsTaskRunner } from './code';
+import { authenticate } from './authenticator';
+
+let _runner: JsTaskRunner;
+
+type Config = {
+	n8nUri: string;
+	authToken: string;
+};
+
+function readAndParseConfig(): Config {
+	const authToken = process.env.N8N_RUNNERS_AUTH_TOKEN;
+	a.ok(authToken, 'Missing task runner auth token. Use N8N_RUNNERS_AUTH_TOKEN to configure it');
+
+	return {
+		n8nUri: 'localhost:5678',
+		authToken,
+	};
+}
+
+void (async function start() {
+	const config = readAndParseConfig();
+
+	const grantToken = await authenticate({
+		authToken: config.authToken,
+		n8nUri: config.n8nUri,
+	});
+
+	const wsUrl = `ws://${config.n8nUri}/rest/runners/_ws?token=${grantToken}`;
```

Comment:
is it safe to pass the grantToken as a query param or could it get intercepted that way, e.g. end up in router logs or so?!

Reply:
In general: no. In this case: yes, because 1) it's single use only 2) n8n should always be deployed in a local cluster so communication should always happen over LAN instead of WAN. I actually had it first as a header, but decided to simplify. I can understand that it might look insecure for some people. Maybe I should move it to a header